pkg/machine/machine_common.go

```diff
@@ -24,3 +27,30 @@ func GetDevNullFiles() (*os.File, *os.File, error) {
 
 	return dnr, dnw, nil
 }
+
+// AddSSHConnectionsToPodmanSocket adds SSH connections to the podman socket if
+// no ignition path is provided
+func AddSSHConnectionsToPodmanSocket(uid, port int, identityPath, name, remoteUsername string, opts InitOptions) error {
```

Comment:
Nit: It's cleaner to return early and inverse the logic.
```golang
if len(opts.IgnitionPath) > 0 {
    fmt.Println("An ignition path was provided.  No SSH connection was added to Podman")
    return nil
}
// Rest of the code
```

It's a rule of thumb that in my experience makes code easier to maintain.